lightclient/src/background.rs

```diff
@@ -69,18 +71,33 @@ pub struct BackgroundTask<TPlatform: PlatformRef, TChain> {
     /// The RPC method request is made in the background and the response should
     /// not be sent back to the user.
     /// Map the request ID of a RPC method to the frontend `Sender`.
-    id_to_subscription: HashMap<
-        (usize, smoldot_light::ChainId),
-        (
-            oneshot::Sender<MethodResponse>,
-            mpsc::UnboundedSender<Box<RawValue>>,
-        ),
-    >,
+    id_to_subscription: HashMap<(usize, smoldot_light::ChainId), SubscriptionIdState>,
     /// Map the subscription ID to the frontend `Sender`.
     ///
     /// The subscription ID is entirely generated by the node (smoldot). Therefore, it is
     /// possible for two distinct subscriptions of different chains to have the same subscription ID.
-    subscriptions: HashMap<(usize, smoldot_light::ChainId), mpsc::UnboundedSender<Box<RawValue>>>,
+    subscriptions: HashMap<(usize, smoldot_light::ChainId), SubscriptionState>,
+}
+
+/// The state needed to resolve the subscription ID and send
+/// back the response to frontend.
+struct SubscriptionIdState {
```

Comment:
I don't like the naming `SubscriptionIdState`, I would prefer `PendingSubscription` or something similar.

Also, would be nice to rename/remove and make a type `ActiveSubscription` that only can happen from the state `PendingSubscription -> ActiveSubscription`

Reply:
Ok, looks like `rejected subscription` are simply just passed to frontend and not inserted in the HashMap.

So, to me it looks like you can merge `SubscriptionState` and `SubscriptionStateId` to single struct and call it Subscription/ActiveSubscription